core/Map.carp

```diff
@@ -123,6 +123,25 @@
     (let [idx (Int.positive-mod (hash k) @(n-buckets m))]
       (Bucket.get (Array.nth (buckets m) idx) k)))
 
+  (doc update "Update value at key k in map with function f, if it exists.")
+  (defn update [m k f]
+    (let [idx (Int.positive-mod (hash k) @(n-buckets &m))]
+      (update-buckets m &(fn [b]
+        (let [n (Array.nth &b idx)]
+          (if (Bucket.contains? n k)
+            (Array.aset b idx (Bucket.put n k &(f (Bucket.get n k))))
```

Comment:
If we want this to be a bit speedier, we could add a function `Bucket.update`; this could reuse the index and thus get rid of an extra lookup.

Reply:
yeah that makes sense

Reply:
ok I managed to whip up something that works, sidesteps the bug, and even looks kinda neat